doc/classes/OS.xml

```diff
@@ -242,7 +242,8 @@
 			<return type="PackedStringArray" />
 			<description>
 				Returns an array of connected MIDI device names, if they exist. Returns an empty array if the system MIDI driver has not previously been initialized with [method open_midi_inputs]. See also [method close_midi_inputs].
-				[b]Note:[/b] This method is implemented on Linux, macOS and Windows.
+				[b]Note:[/b] This method is implemented on Linux, macOS, Web, and Windows.
+				[b]Note:[/b] On the Web platform, using MIDI input requires a browser permission to be granted first. This permission request is performed when calling [method open_midi_inputs]. MIDI input will not work until the user accepts the permission request.
```

Comment:
I don't feel like it's necessary to note it here, because the prior description already says that you need to call [method open_midi_inputs] for this method to work (which contains more details).

If the driver has not been initialized on Web due to lack of permissions, it won't work. The logic feels straightforward.

Reply:
> What isn't straightforward is the asynchronous nature of it, I assume?

Yes, exactly.

> On Web, are there any chances that the array may "falsely" return empty? If yes, that's worth noting down.

Apart from the async nature, and the permissions request being denied, the other "false" cases would be:

 1. The midi device must be connected at the time of calling `open_midi_inputs` the first time - since the callbacks are cached after that. I could start listening to [the statechange event](https://developer.mozilla.org/en-US/docs/Web/API/MIDIAccess/statechange_event), and then the array of connected inputs would change over time, but it doesn't look like any of the other MIDI drivers (alsamidi/coremidi/winmidi) do that? For example, every time you receive an `InputEventMIDI`, the `device` would need to be checked against the current `OS.get_connected_midi_inputs()` if you wanted to know which device it came from.
    * (re: "the first time") As a "workaround", you can "refresh" the inputs by calling `close_midi_inputs()` and then `open_midi_inputs()` again.
 2. Apart from the permissions request being denied, these are the other limitations called out on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Web_MIDI_API#security_requirements):
    1. The method must be called in a [secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts).
       * But I don't think we need to call this out as being a special requirement for MIDI on web.
    2. Access may be gated by the [midi](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy/midi) HTTP [Permission Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Permissions_Policy).
       * Maybe this would be too obscure to note here? The [default permissions policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy/midi) is that `midi` would be allowed, so whoever is hosting the godot web app would have specifically blocked it for you to run into it.
  3. The user's browser not supporting WebMIDI.
      * [Mostly Safari, and Firefox for Android](https://caniuse.com/?search=webmidi)